core/src/main/java/org/apache/accumulo/core/metadata/ValidationUtil.java

```diff
@@ -53,6 +53,7 @@ public static Path validate(Path path) {
 
   public static void validateRFileName(String fileName) {
     Objects.requireNonNull(fileName);
+    // TODO: In 3.0.0 validate that filename starts with FilePrefix
```

Comment:
You should be able to implement this todo here and remove the comment. Something like `FilePrefix.fromPrefix(fileName.substring(0,1));`